app/models/blacklight_config.rb

```diff
@@ -0,0 +1,27 @@
+class BlacklightConfig
+  attr_reader :metadata_class
+  def initialize(metadata_class)
+    @metadata_class = metadata_class
+    apply_show_fields
+  end
+
+  def default_configuration
+    CatalogController.blacklight_config.deep_copy
```

Comment:
Actually I think I just realized how this works - the controller instantiates the model and requests its configuration which is done by asking the controller for its configuration, ...

How is this not causing stack errors?